src/routes/register/+page.svelte

```
@@ -0,0 +1,349 @@
+			onUsernameChanged();
+		}, 1000);
+	}
+
+	function startTimerEmail() {
+		if (timerEmail !== null) {
+			clearTimeout(timerEmail);
+			timerEmail = null;
+		}
+		timerEmail = setTimeout(() => {
+			onEmailChanged();
+		}, 1000);
+	}
+
+	function samePassword() {
+		return password_1.trim() === password_2.trim();
+	}
+
+	async function onUsernameChanged() {
+		const reset = () => {
+			usernameMessage = '';
+		};
+
+		if (username.trim() === '') {
+			reset();
+			return;
+		}
+
+		const response = await fetch('api/account/username/exists?username=' + username);
+		if (!response.ok) {
+			usernameMessage = 'Fehler beim überprüfen des Namen. Fehlercode: ' + response.status;
+			return;
+		}
+
+		let data;
+		try {
+			data = await response.json();
+		} catch (error) {
+			console.error('error while parsing username json', error);
+			reset();
+			return;
+		}
+
+		if (data.exists) {
+			usernameMessage = 'Der Name  "' + username + '" ist bereits vergeben.';
+			return;
+		}
+
+		reset();
+	}
+
+	async function onEmailChanged() {
+		const reset = () => {
+			emailMessage = '';
+		};
+
+		if (email.trim() === '') {
+			reset();
+			return;
+		}
+
+		const response = await fetch('api/account/email/exists?email=' + email);
+		if (!response.ok) {
+			emailMessage = 'Fehler beim überprüfen der E-Mail. Fehlercode: ' + response.status;
+			return;
+		}
+
+		let data;
+		try {
+			data = await response.json();
+		} catch (error) {
+			console.error('error while parsing email json', error);
+			reset();
+			return;
+		}
+
+		if (data.exists) {
+			emailMessage = 'Die E-Mail "' + email + '" wird bereits verwendet.';
+			return;
+		}
+
+		reset();
+	}
+
+	function onPasswordChanged() {
+		const reset = () => {
+			passwordMessage = '';
+		};
+
+		if (password_1.trim() === '') {
+			reset();
+			return;
+		}
+
+		if (password_2.trim() === '') {
+			reset();
+			return;
+		}
+
+		if (!samePassword()) {
+			passwordMessage = 'Die Passwörter stimmen nicht überein.';
```

When registering, the error message "Die Passwörter stimmen nicht überein." appears while typing the repetition of the password. I find this a bit annoying. Should we maybe change it so that the message only appears if the repeated password is at least as long as the original password?

---

Did it even have a timeout before? I can't remember. However, I think the current timeout is too long. It takes too much time until the error message appears. Also, another problem is that if I correct the error, it takes a long time for the error message to disappear. That can be confusing.
I don't really know if the timeout is the right way here. At least not in the general case. I'm thinking of a solution that depends on the actual input, e.g.:
- Is the repeated password shorter than the original password? → Use a timeout (~ 1 sec?)
- Is it as long as or even longer than the original? → Zero timeout, show message immediately
- Does it match the original password? → Hide all error messages immediately

Of course, I don't know how difficult this is to build.

---

I can try that. Well see. But this will result in a custom function. Later on will be the general call.

---

I now have a function that sets the timer time based on the password length. What do you thin?

---

I think it's way better than before👍 But, if the second password is not shorter than the first one, why do you use a timeout of 200 ms? You could display the error immediately then because you are absolutely sure that the user has done something wrong. So why still wait?